pallet/staking/src/lib.rs

```diff
@@ -901,24 +891,24 @@ pub mod pallet {
 		///
 		/// This should only be called by the [`pallet_session::SessionManager::new_session`].
 		pub fn elect() -> Vec<T::AccountId> {
-			let mut collators = <Collators<T>>::iter_keys()
-				.map(|c| {
-					let mut t_power = 0;
-					let i_exposures = <Nominators<T>>::iter()
+			let mut collators = <Collators<T>>::iter()
+				.map(|(c, cm)| {
+					let mut collator_v = 0;
+					let nominators = <Nominators<T>>::iter()
 						.filter_map(|(n, c_)| {
 							if c_ == c {
-								let n_power = Self::power_of(&n);
+								let nominator_v = (Perbill::one() - cm) * Self::power_of(&n);
 
-								t_power += n_power;
+								collator_v += nominator_v;
 
-								Some(IndividualExposure { who: n, value: n_power })
+								Some(IndividualExposure { who: n, vote: nominator_v })
 							} else {
 								None
 							}
 						})
 						.collect();
 
-					((c, Exposure { total: t_power, nominators: i_exposures }), t_power)
+					((c, Exposure { vote: collator_v, nominators }), collator_v)
```

Comment:
One small suggestion for improvement: 
You can calculate the total_power first and then mul the `(Perbill::one() - cm)` later, in this way reducing the mul times.

Reply:
Nope, `nomonator_v` also need to be scaled. But I can take the `Perbill::one - cm` out of the loop.